web/reqlatencycounter.go

```diff
@@ -0,0 +1,63 @@
+package web
+
+import (
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/timekeeper"
+)
+
+// FastRequestLimitDuration is the interface for getting the cutoff time for bad requests.
+type FastRequestLimitDuration interface {
+	Get() time.Duration
+}
+
+// ReqLatencyCounter hold static data for the request latency tracking.
+type ReqLatencyCounter struct {
+	fastRequestLimitDuration FastRequestLimitDuration
```

Comment:
Will probably be difficult to use by others if ```fastRequestLimitDuration``` is private unless you provide a constructor.